sunspot_rails/lib/sunspot/rails/searchable.rb

```diff
@@ -98,10 +98,17 @@ def searchable(options = {}, &block)
             end
 
             unless options[:auto_remove] == false
+              on_filter_invalid = /(before|after|around)_(save|destroy|create)/
```

Comment:
I guess `(before|after|around)` is the redundant part? For me this kind of explicitness makes it difficult to read

Reply:
Yeah `on` isn't valid for the create/save/destroy callbacks. It used to be ignored and now raises an error. I can drop the first half of the regex and just check for those strings instead if you prefer.